AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/functionsFlow.js

```diff
@@ -267,27 +265,36 @@ function getMaxCheckLevel(subChecks) {
     return maxCheckLevel;
 }
 
-async function networkCheckConnectionString(propertyName, connectionString, connectionStringType = undefined, dnsServers, diagProvider, isVnetIntegrated, failureDetailsMarkdown = undefined) {
+async function networkCheckConnectionString(propertyName, connectionString, connectionStringType = undefined, dnsServers, diagProvider, isVnetIntegrated, failureDetailsMarkdown = undefined, entityName=undefined) {

```

Comment:
Let's simplify the code here now that we have changed our logic to rely on app setting entirely.  Thought we discussed this early on in the design.  Remove the "connectionString" parameter here and just do a lookup based on the propertyName.  Remove the few lines of code above this that lookup the connection string from app setting (i.e. for WEBSITE_RUN_FROM_PACKAGE and APPLICATIONINSIGHTS_CONNECTION_STRING) and just pass the appSettings dictionary.  We don't get to reduce the number of parameters but we simplify the code above to make it 100% app settings based instead of a mix of that and connection string.  The existing logic is a bit messy and could be error prone in future maintenance of this code.

Reply:
In case of APPLICATIONINSIGHTS_CONNECTION_STRING -- the validation is done using the connection string via TCP Ping. It does not go to the DaaS. So, the networkCheckConnectionString function requires the connectionstring parameter. But we have removed this parameter from the validateConnectionViaAppSetting